### DISCUSSION TITLE
Apollo Router vs. Cosmo query plan execution?

### DISCUSSION THREAD

**Discussion by rhino88** [wundergraph/cosmo#217] [Q&A] [answered]
Is the expectation that the Cosmo router will have exact parity with the Apollo router? 

For example, we have discovered [an overfetching issue](https://github.com/apollographql/federation/issues/2759) in the Apollo router which significantly increases the latency of some of our federated queries in production. 

We have confirmed that this issue also exists in cosmo/wundergraph. Would this be something that Apollo needs to fix before Cosmo would do the same?

**Comment by devsergiy** [comment, marked as answer]
Hi @rhino88 

We are not exposing yet human readable query plans

But there is a way to print internal plans by providing this environment variable

```
 ENGINE_DEBUG_PRINT_QUERY_PLANS=true
```

Could you run the router with your configuration and paste query plans here?

**Reply by rhino88** [reply]
Sure thing, I also realized that I created the example using the [Apollo Federation quickstart](https://docs.wundergraph.com/docs/examples/apollo-federation) which does not use cosmo. Will publish an example asap.
